cli/internal/util/backends.go

```diff
@@ -22,43 +23,19 @@ func IsYarn(backendName string) bool {
 
 func IsBerry(cwd string, version string, pkgManager bool) (bool, error) {
 	if pkgManager {
-		v, err := semver.NewVersion(version)
-		if err != nil {
-			return false, fmt.Errorf("could not parse yarn version: %w", err)
-		}
-		c, err := semver.NewConstraint(">=2.0.0")
+		version, err := strconv.Atoi(strings.SplitN(version, ".", 2)[0])
 		if err != nil {
-			return false, fmt.Errorf("could not create constraint: %w", err)
+			return false, err
 		}
 
-		return c.Check(v), nil
-	} else {
-		if !commandExists("yarn") {
-			return false, nil
-		}
-		cmd := exec.Command("yarn", "--version")
-		cmd.Dir = cwd
-		out, err := cmd.Output()
-		if err != nil {
-			return false, fmt.Errorf("could not detect yarn version: %w", err)
+		if version >= 2 {
+			return true, nil
 		}
 
-		v, err := semver.NewVersion(strings.TrimSpace(string(out)))
-		if err != nil {
-			return false, fmt.Errorf("could not parse yarn version: %w", err)
-		}
-		c, err := semver.NewConstraint(">=2.0.0")
-		if err != nil {
-			return false, fmt.Errorf("could not create constraint: %w", err)
-		}
-
-		return c.Check(v), nil
+		return false, nil
 	}
-}
 
-func commandExists(cmd string) bool {
-	_, err := exec.LookPath(cmd)
-	return err == nil
+	return fs.PathExists(filepath.Join(cwd, ".yarnrc.yml")), nil
```

Comment:
Is `.yarnrc.yml` actually required to exist? I know in yarn 2 they changed this:
> Starting from the v2, they must be written in valid Yaml and have the right extension (simply calling your file .yarnrc won't do).

As as yarn 1 repo's might have a .yarnrc, but I think even that isn't really guaranteed to exist.

Reply:
This is only to check for Yarn V2 and up, Yarn V1 doesn't use `.yarnrc.yml`.

Reply:
I think I misread the diff and thought this was still a more general function. My question was originally more about yarn 2 not requiring a `.yarnrc.yml` file either, but I think that is just your fallback in case the package.json doesn't specify the `packageManager` field?

Reply:
Yeah, even though it's a fallback I think Yarn V2 and up still need the file, so it should always be there regardless.

Reply:
> I think Yarn V2 and up still need the file

It doesn't, there is no requirements for it to exist.